chat-plugins/room-events.js

```diff
@@ -14,64 +14,84 @@ exports.commands = {
 	roomevent: 'roomevents',
 	roomevents: {
 		'': function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
-			if (!room.events || Object.keys(room.events).length === 0) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!room.events || !Object.keys(room.events).length) {
 				return this.errorReply("There are currently no planned upcoming events for this room.");
 			}
 			if (!this.runBroadcast()) return;
 			let buff = '<table border="1" cellspacing="0" cellpadding="3">';
 			buff += '<th>Event Name:</th><th>Event Description:</th><th>Event Date:</th>';
-			let events = Object.keys(room.events);
-			for (let i = 0; i < events.length; i++) {
-				buff += `<tr><td>${Chat.escapeHTML(events[i])}</td><td>${room.events[events[i]].desc}</td><td><time>${Chat.escapeHTML(room.events[events[i]].date)}</time></td></tr>`;
+			for (let i in room.events) {
+				buff += `<tr><td>${Chat.escapeHTML(room.events[i].eventName)}</td><td>${Chat.parseText(room.events[i].desc)}</td><td><time>${Chat.escapeHTML(room.events[i].date)}</time></td></tr>`;
 			}
 			buff += '</table>';
 			return this.sendReply(`|raw|<div class="infobox-limited">${buff}</div>`);
 		},
 		add: function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
 			if (!this.can('declare', null, room)) return false;
 			if (!room.events) room.events = {};
-			let [eventName, desc, ...date] = target.split('|');
-			date = date.join("|");
-			if (!eventName || !desc || !date) return this.errorReply("You're missing a command parameter - see /help roomevents for this command's syntax.");
-			desc = this.canHTML(desc);
-			if (desc === false) return false; // HTML issue - line above will be more specific
+			let [eventName, date, ...desc] = target.split('|');
+
+			if (!(eventName && date && desc)) return this.errorReply("You're missing a command parameter - see /help roomevents for this command's syntax.");
+
 			eventName = eventName.trim();
-			if (room.events[eventName]) return this.errorReply("An event with this name is already added to the events list.");
-			room.events[eventName] = {
-				desc: desc,
+			date = date.trim();
+			desc = desc.join('|').trim();
+
+			if (eventName.length > 50) return this.errorReply("Event names should not exceed 50 characters.");
+			if (date.length > 150) return this.errorReply("Event dates should not exceed 150 characters.");
+			if (desc.length > 1000) return this.errorReply("Event descriptions should not exceed 1000 characters.");
+
+			let eventId = toId(eventName);
```

Comment:
Also, if you aren't changing the value of ``eventId``, then this should probably use ``const``